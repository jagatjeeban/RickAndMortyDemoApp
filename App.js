import React from "react";
import { StyleSheet, StatusBar } from "react-native";
import FlashMessage from "react-native-flash-message";

//import main stack navigator
import MainStackNavigator from "./src/navigators";

//import constants
import { Colors } from "./src/common/constants";

const App = () => {
  return(
    <>
      <StatusBar backgroundColor={Colors.BgColor} />
      <MainStackNavigator />
      <FlashMessage position={'bottom'} floating />
    </>
  )
}

export default App;

const styles = StyleSheet.create({
  safeAreaView: {
    flex: 1,
    alignItems:'center',
    justifyContent:'center',
    backgroundColor: 'white'
  }
})