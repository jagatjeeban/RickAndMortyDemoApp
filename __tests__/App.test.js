import 'react-native';
import React from 'react';
import App from '../App';

//import render 
import { render } from '@testing-library/react-native';

// Note: import explicitly to use the types shipped with jest.
import {it} from '@jest/globals';

// Mocking react-native-flash-message
jest.mock('react-native-flash-message', () => 'FlashMessage');

// Mocking react-navigation stack navigator
jest.mock('../src/navigators', () => {
  return {
    __esModule: true,
    default: () => {
      return <></>;
    },
  };
});

it('app renders correctly', () => {
  const snapshot = render(<App/>).toJSON();
  expect(snapshot).toMatchSnapshot();
});
