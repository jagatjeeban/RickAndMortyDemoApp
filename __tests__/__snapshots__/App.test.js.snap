// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`app renders correctly 1`] = `
<FlashMessage
  floating={true}
  position="bottom"
/>
`;
